elements/lisk-p2p/src/peer_book/base_list.ts

```diff
@@ -13,146 +13,192 @@
  *
  */
 import { ExistingPeerError } from '../errors';
-import { P2PPeerInfo } from '../p2p_types';
-import { evictPeerRandomlyFromBucket, getBucketId, PEER_TYPE } from '../utils';
+import { P2PEnhancedPeerInfo, P2PPeerInfo } from '../p2p_types';
+import {
+	evictPeerRandomlyFromBucket,
+	getBucketId,
+	PEER_TYPE,
+	sanitizeInternalPeerInfo,
+} from '../utils';
 
 export interface PeerListConfig {
-	readonly peerBucketCount: number;
-	readonly peerBucketSize: number;
+	readonly numOfBuckets: number;
+	readonly bucketSize: number;
 	readonly secret: number;
 	readonly peerType: PEER_TYPE;
 }
 
-export interface CustomPeerInfo {
-	readonly peerInfo: P2PPeerInfo;
-	readonly dateAdded: Date;
+export type Bucket = Map<string, P2PEnhancedPeerInfo>;
+
+export interface BucketInfo {
```

Comment:
This type looks weird. I was expecting a map of buckets identified by its ids. Do we really need it?

Reply:
This is the return type to `calculateBucket` and is required to get `bucket` based on only a `peerId`. See `getBucket` function below.